app/views/robot_verifications/submission.html.erb

```diff
@@ -16,55 +16,72 @@
     </tr>
   </thead>
   <tbody>
-  <% if @all_labels.last.nil? -%>
+  <% if @source_plates.blank? -%>
     <tr><td>No source plates found for this batch</td></tr>
   <% else -%>
-    <% count = 0%>
```

Comment:
Switch to each with index and simplify sort. Although given this sort is happening all over the place, it probably indicates we should sort the list upfront.